### PR TITLE
fix(webpack-cli): verbose and stats verbose flag fixed (rebase)

### LINKED ISSUE #1443: Bug: webpack-cli --verbose throws TypeError: Cannot set property 'stats' of undefined
**Describe the bug**

**What is the current behavior?**
`webpack-cli --verbose` or `webpack-cli --verbose --stats verbose` throws an error described below.
```javascript
[webpack-cli] TypeError: Cannot set property 'stats' of undefined
    at StatsGroup.resolveOptions (/home/phoenix/Desktop/webpack-cli/packages/webpack-cli/lib/groups/StatsGroup.js:23:40)
    at StatsGroup.run (/home/phoenix/Desktop/webpack-cli/packages/webpack-cli/lib/groups/StatsGroup.js:33:14)
    at WebpackCLI._handleGroupHelper (/home/phoenix/Desktop/webpack-cli/packages/webpack-cli/lib/webpack-cli.js:220:40)
    at Promise.resolve.then.then.then.then.then.then.then (/home/phoenix/Desktop/webpack-cli/packages/webpack-cli/lib/webpack-cli.js:269:30)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:745:11)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
```

**To Reproduce**

Use either verbose flag or verbose flag with some value of stats flag. 

**Expected behavior**
Using verbose flags should output everything. If the verbose flag is used with stats flag then it should also output everything if the stats has a valid flag value else, it should also throw a warning if the stats flag value is not a valid one.

**Screenshots**
Screenshot from 2020-04-11 18-56-15

**Please paste the results of `webpack-cli info` here, and mention other relevant information**
```bash
System:
    OS: Linux 4.19 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (4) x64 Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
    Memory: 693.40 MB / 7.66 GB
  Binaries:
    Node: 10.15.2 - /usr/bin/node
    Yarn: 1.22.4 - /usr/bin/yarn
    npm: 5.8.0 - /usr/bin/npm
  Browsers:
    Chrome: 80.0.3987.149
    Firefox: 68.6.0esr
  npmPackages:
    webpack: ^4.42.0 => 5.0.0-beta.14 
    webpack-cli: ^3.3.11 => 4.0.0-beta.8 
```
**Additional context**

### PR DESCRIPTION
**Pull request by aman29271**
**What kind of change does this PR introduce?**
This is a bugfix request which closes #1443 

**Did you add tests for your changes?**
yes
**If relevant, did you update the documentation?**
Not Needed
**Summary**

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

**Comment by aman29271** [comment]
Oh, Ah. I forgot that. working on it...

File: packages/webpack-cli/lib/groups/StatsGroup.js [hunk 1]
```diff
@@ -5,7 +5,9 @@ const logger = require('../utils/logger');
  */
 class StatsGroup extends GroupHelper {
     static validOptions() {
-        return ['none', 'errors-only', 'minimal', 'normal', 'detailed', 'verbose', 'errors-warnings'];
+        let validArrayString = ['none', 'errors-only', 'minimal', 'normal', 'detailed', 'verbose', 'errors-warnings'];
+        let validArrayObject = [{ verbose: true }, { verbose: true, stats: 'verbose' }];
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/groups/StatsGroup.js:9]: Why we need it here? We don't support objects in webpack-cli

Comment (aman29271) [reply packages/webpack-cli/lib/groups/StatsGroup.js:9]: I used that for tests. It deals with when a user provide `--stats verbose --verbose` both flags together.

File: packages/webpack-cli/lib/utils/cli-flags.js [hunk 2]
```diff
@@ -9,6 +9,12 @@ const ADVANCED_GROUP = 'advanced';
 const DISPLAY_GROUP = 'stats';
 const ZERO_CONFIG_GROUP = 'zero-config';
 
+function acceptedString(arr) {
+    return `Accepted Value: ${arr.join(' | ')}`;
+}
+function descriptionGenerator(example, shortDesc, acceptedValue = []) {
+    return `\n\n    Example: ${example}\n\n     ${acceptedValue.length > 0 ? acceptedString(acceptedValue) : ''}\n\n${shortDesc}`;
+}
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/utils/cli-flags.js:17]: It was fix, not new feature, do not mix fixes and feature, you need to do it in difference PRs

Comment (aman29271) [reply packages/webpack-cli/lib/utils/cli-flags.js:17]: Okay. On it...

File: packages/webpack-cli/lib/groups/StatsGroup.js (hunk 1 above)
Comment (evilebottnawi) [reply packages/webpack-cli/lib/groups/StatsGroup.js:9]: Put it in tests - you can sue mocking

**Pull request closed without merging**